src/components/AnalysisModal.js

```diff
@@ -75,10 +87,10 @@ export const AnalysisModal = Utils.withDisplayName('AnalysisModal')(
         }],
         [Utils.DEFAULT, () => Utils.cond(
           [currentTool === tools.RStudio.label || currentTool === tools.Jupyter.label, () => setViewMode(analysisMode)],
-          [currentTool === tools.galaxy.label && !currentApp, () => setViewMode(environmentMode)],
-          [currentTool === tools.galaxy.label && currentApp, () => {
+          [isToolAnApp(currentTool) && !app, () => setViewMode(environmentMode)],
+          [isToolAnApp(currentTool) && app, () => {
```

Comment:
When we use non-boolean values in logical operations as truthy, we typically prefix them with `!!` to make the intention clear. `truthy/falsy` concepts can be fairly foreign for people coming from non-Javascript backgrounds.

```suggestion
          [isToolAnApp(currentTool) && !!app, () => {
```